mule-extensions-api/src/main/java/org/mule/runtime/extension/internal/loader/validator/NameClashModelValidator.java

```diff
@@ -596,6 +598,20 @@ public String toString() {
     }
   }
 
+  private boolean areTypesEqual(MetadataType type, MetadataType otherType) {
```

Comment:
it'd be good to have this moved to `org.mule.runtime.extension.api.util.ExtensionMetadataTypeUtils` instead